service/history/shard/context_impl.go

```diff
@@ -374,19 +375,21 @@ func (s *ContextImpl) GetQueueState(
 
 func (s *ContextImpl) SetQueueState(
 	category tasks.Category,
+	tasksCompleted int,
 	state *persistencespb.QueueState,
 ) error {
-	return s.updateShardInfo(func() {
-		categoryID := category.ID()
-		s.shardInfo.QueueStates[int32(categoryID)] = state
-	})
+	return s.updateShardInfo(tasksCompleted,
+		func() {
+			categoryID := category.ID()
+			s.shardInfo.QueueStates[int32(categoryID)] = state
+		})
 }
 
 func (s *ContextImpl) UpdateReplicationQueueReaderState(
 	readerID int64,
 	readerState *persistencespb.QueueReaderState,
 ) error {
-	return s.updateShardInfo(func() {
+	return s.updateShardInfo(0, func() {
```

Comment:
nit: Add a TODO here? 

cc @yux0 do you think this is useful for replication queue?

Reply:
Yeah, that's a TODO before I merge this. I planned to revisit it after @yux0 taught all of us new folk about our replication design